src/components/shared/legend.js

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { Input, Select, Row, Col } from 'antd';
+import { useDispatch, useSelector } from 'react-redux';
+
+import _ from "lodash";
+
+const { Option } = Select;
+
+function Legend() {
+	const spec = useSelector(state => state.chart.spec);
+	const layer = spec.layer[0];
+	const {title, fillColor, symbolType, symbolSize, orient} = layer.encoding.color.legend;
```

Comment:
TypeError: Cannot destructure property 'title' of 'layer.encoding.color.legend' as it is undefined.